ledger/db/src/lib.rs

```diff
@@ -279,46 +292,68 @@ impl LedgerDB {
     #[allow(clippy::unreadable_literal)]
     pub fn open(path: PathBuf) -> Result<LedgerDB, Error> {
         let env = Environment::new()
-            .set_max_dbs(21)
+            .set_max_dbs(22)
             .set_map_size(MAX_LMDB_FILE_SIZE)
             // TODO - needed because currently our test cloud machines have slow disks.
             .set_flags(EnvironmentFlags::NO_SYNC)
             .open(&path)?;
 
-        // Check if the database we opened is compatible with the current implementation.
         let metadata_store = MetadataStore::new(&env)?;
-        let db_txn = env.begin_ro_txn()?;
-        let version = metadata_store.get_version(&db_txn)?;
-        global_log::info!("Ledger db is currently at version: {:?}", version);
-        db_txn.commit()?;
 
-        match version.is_compatible_with_latest() {
-            Ok(_) => {}
-            // Version 20200610 introduced the TxOut public key -> index store.
-            Err(Error::VersionIncompatible(20200427, 20200610)) => {
-                global_log::info!("Ledger db migrating from version 20200427 to 20200610, this might take awhile...");
-
-                TxOutStore::construct_tx_out_index_by_public_key_from_existing_data(&env)?;
-
-                let mut db_txn = env.begin_rw_txn()?;
-                metadata_store.set_version_to_latest(&mut db_txn)?;
-                global_log::info!(
-                    "Ledger db migration complete, now at version: {:?}",
-                    metadata_store.get_version(&db_txn),
-                );
-                db_txn.commit()?;
-            }
-            Err(err) => {
-                return Err(err);
-            }
-        };
+        loop {
```

Comment:
Thank you for this! To check my understanding - the loop makes it so that it can update to 20200610 first, then 20200707?

Reply:
Yup